File: srcnew/DesktopConfig.pas
unit DesktopConfig;

{$mode objfpc}{$H+}

interface

uses
  SysUtils, fgl;

type
  TDesktopInfo = class
  private
    FId: Integer;
    FName: string;
    FPath: string;
    FIsDefault: Boolean;
  public
    constructor Create; overload;
    /// 创建"当前桌面"实例（不可删除，路径运行时动态获取）
    constructor CreateDefault;
    /// 创建新桌面实例
    constructor CreateNew(AId: Integer; const ABaseName: string = '新桌面');
    /// 获取实际桌面路径：默认桌面返回系统桌面路径
    function GetEffectivePath: string;
    function ToString: string; override;
    property Id: Integer read FId write FId;
    property Name: string read FName write FName;
    property Path: string read FPath write FPath;
    property IsDefault: Boolean read FIsDefault write FIsDefault;
  end;

  TDesktopInfoList = specialize TFPGObjectList<TDesktopInfo>;

  TDesktopConfigData = class
  private
    FFirstRun: Boolean;
    FDesktops: TDesktopInfoList;
  public
    constructor Create; overload;
    /// 创建初始配置（含一个"当前桌面"）
    constructor CreateDefault;
    destructor Destroy; override;
    property FirstRun: Boolean read FFirstRun write FFirstRun;
    property Desktops: TDesktopInfoList read FDesktops;
  end;

implementation

// ===== TDesktopInfo =====

constructor TDesktopInfo.Create;
begin
  inherited;
  FId := 0;
  FName := '';
  FPath := '';
  FIsDefault := False;
end;

constructor TDesktopInfo.CreateDefault;
begin
  Create;
  FId := 0;
  FName := '当前桌面';
  FPath := '';
  FIsDefault := True;
end;

constructor TDesktopInfo.CreateNew(AId: Integer; const ABaseName: string);
begin
  Create;
  FId := AId;
  FName := ABaseName + IntToStr(AId);
  FPath := '';
  FIsDefault := False;
end;

function TDesktopInfo.GetEffectivePath: string;
begin
  if FIsDefault then
    Result := GetEnvironmentVariable('USERPROFILE') + '\Desktop'
  else
    Result := FPath;
end;

function TDesktopInfo.ToString: string;
begin
  Result := FName;
end;

// ===== TDesktopConfigData =====

constructor TDesktopConfigData.Create;
begin
  inherited;
  FFirstRun := True;
  FDesktops := TDesktopInfoList.Create(True);  // OwnsObjects = True
end;

constructor TDesktopConfigData.CreateDefault;
begin
  Create;
  FFirstRun := True;
  FDesktops.Add(TDesktopInfo.CreateDefault);
end;

destructor TDesktopConfigData.Destroy;
begin
  FDesktops.Free;
  inherited;
end;

end.

File: src/MultiDesktop/Models/DesktopConfig.cs
using System;
using System.Collections.Generic;

namespace MultiDesktop.Models
{
    /// <summary>
    /// 单个桌面的配置信息
    /// </summary>
    public class DesktopInfo
    {
        /// <summary>唯一标识</summary>
        public int Id { get; set; }

        /// <summary>桌面名称</summary>
        public string Name { get; set; }

        /// <summary>桌面文件夹路径</summary>
        public string Path { get; set; }

        /// <summary>是否为当前桌面（不可删除）</summary>
        public bool IsDefault { get; set; }

        /// <summary>获取实际桌面路径：当前桌面始终返回系统当前桌面路径</summary>
        public string GetEffectivePath()
        {
            if (IsDefault)
                return Environment.GetFolderPath(Environment.SpecialFolder.Desktop);
            return Path;
        }

        public DesktopInfo()
        {
            Id = 0;
            Name = "";
            Path = "";
            IsDefault = false;
        }

        /// <summary>创建当前桌面实例（路径不存 JSON，运行时动态读取）</summary>
        public static DesktopInfo CreateDefault()
        {
            return new DesktopInfo
            {
                Id = 0,
                Name = "当前桌面",
                Path = "",
                IsDefault = true
            };
        }

        /// <summary>创建新桌面实例</summary>
        public static DesktopInfo CreateNew(int id, string baseName = "新桌面")
        {
            return new DesktopInfo
            {
                Id = id,
                Name = baseName + id,
                Path = "",
                IsDefault = false
            };
        }

        public override string ToString()
        {
            return Name;
        }
    }

    /// <summary>
    /// 桌面配置文件根对象
    /// </summary>
    public class DesktopConfigData
    {
        /// <summary>是否首次运行</summary>
        public bool FirstRun { get; set; }

        /// <summary>所有桌面列表</summary>
        public List<DesktopInfo> Desktops { get; set; }

        public DesktopConfigData()
        {
            FirstRun = true;
            Desktops = new List<DesktopInfo>();
        }

        /// <summary>创建当前配置（首次运行）</summary>
        public static DesktopConfigData CreateDefault()
        {
            var config = new DesktopConfigData
            {
                FirstRun = true
            };
            config.Desktops.Add(DesktopInfo.CreateDefault());
            return config;
        }
    }
}
